packages/analytics-browser/src/plugins/destination.ts

```diff
@@ -0,0 +1,17 @@
+import { Destination as CoreDestination, buildResult } from '@amplitude/analytics-core';
+import { BrowserConfig, DestinationContext as Context } from '@amplitude/analytics-types';
+
+export class Destination extends CoreDestination {
```

Comment:
why do we need a subclass?

Reply:
This project only includes the Browser SDK so I want to minimize the impact on other SDKs. That's why I only add the `diagnosticProvider` property to `BrowserConfig` interface rather than `Config` interface in core package. wdyt?